test/util/server/server.go

```diff
@@ -395,6 +401,46 @@ func serviceAccountSecretsExist(client *kclient.Client, namespace string, sa *ka
 	return foundTokenSecret && foundDockercfgSecret
 }
 
+// WaitForPodCreationServiceAccounts ensures that the service account needed for pod creation exists
+// and that the cache for the admission control that checks for pod tokens has caught up to allow
+// pod creation. A test pod can be passed to test admission creation. If nil, a default pod is used
+// for the test.
+func WaitForPodCreationServiceAccounts(client *kclient.Client, namespace string, serviceAccounts []string, testPod *kapi.Pod) error {
+	err := WaitForServiceAccounts(client, namespace, serviceAccounts)
+	if err != nil {
+		return err
+	}
+	if testPod == nil {
+		testPod = &kapi.Pod{}
+		testPod.GenerateName = "test"
+		testPod.Spec.Containers = []kapi.Container{
+			{
+				Name:  "container",
+				Image: "test/image",
```

Comment:
Can you choose an image that is likely to be present on the node already or will pull very quickly?  Its possible that someone will lift this for our e2e which actually has a node.